src/components/Calculator.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import NumberButtons from './NumberButtons';
+import OperatorButtons from './OperandButtons';
+import { expressionStorage } from './store/store';
+import {
+  NUMBER_LIMIT,
+  ERROR_MSG,
+  CONFIRM_MSG,
+  OPERATOR,
+  OPERATOR_LIST,
+} from './constants/constant';
+import AllClearButton from './AllCearButton';
+import Screen from './Screen';
+class Calculator extends Component {
+  constructor() {
+    super();
+    window.addEventListener('beforeunload', this.confirmExist);
+  }
+  state = {
+    sum: '',
+    prevNumber: [],
+    operator: '',
+    nextNumber: [],
+  };
+
+  componentDidMount() {
+    const expression = expressionStorage.getExpression();
+    if (!expression) {
+      return;
+    }
+    const { sum, prevNumber, operator, nextNumber } = expression;
+    this.setState({ sum, prevNumber, operator, nextNumber });
+  }
+
+  confirmExist = (e) => {
+    e.preventDefault();
+    e.returnValue = CONFIRM_MSG;
+    const { sum, prevNumber, operator, nextNumber } = this.state;
+    expressionStorage.setExpression({ sum, prevNumber, operator, nextNumber });
```

Comment:
```suggestion
    expressionStorage.setExpression({ ...this.state });
```
spread operator를 써보면 어떨까요?

Reply:
편한 길을 두고 돌아가고 있었네요😅  ... 을 활용해서 수정해보겠습니다!